File: ago-dic-2023/Darwin Solis/Practica 1/Suma.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 */

package com.mycompany.suma;
import java.util.Scanner;
/**
 *
 * @author Warwin02
 */
public class Suma {

    public static void main(String[] args) {
         Scanner scanner = new Scanner(System.in);
        
        System.out.print("Ingrese dos números separados por un espacio: ");
        
        // Leer la entrada del usuario como una cadena de texto
        String entrada = scanner.nextLine();
        
        // Dividir la entrada en dos partes utilizando el espacio como separador
        String[] partes = entrada.split(" ");
        
        // Verificar si se ingresaron dos números
        if (partes.length != 2) {
            System.out.println("Debe ingresar exactamente dos números separados por un espacio.");
        } else {
            try {
                // Convertir las partes en números
                double numero1 = Double.parseDouble(partes[0]);
                double numero2 = Double.parseDouble(partes[1]);
                
                // Calcular la suma de los números
                double suma = numero1 + numero2;
                
                System.out.println("La suma de los números es: " + suma);
            } catch (NumberFormatException e) {
                System.out.println("Error: Ingrese dos números válidos separados por un espacio.");
            }
        }
        
        scanner.close();
    
    }
}

File: ago-dic-2024/Zulema/Suma.py
def main():
    while True:
        # Solicitar la entrada del usuario
        entrada = input("Introduce dos números enteros separados por un espacio (o escribe 'salir' para terminar): ")
        
        # Comprobar si el usuario quiere salir
        if entrada.lower() == 'salir':
            print("Programa terminado.")
            break
        
        # Dividir la entrada en partes
        partes = entrada.split()
        
        # Verificar que haya exactamente dos partes
        if len(partes) == 2:
            try:
                # Convertir las partes a enteros
                a = int(partes[0])
                b = int(partes[1])
                
                # Calcular la suma
                suma = a + b
                
                # Imprimir el resultado
                print(f"La suma es: {suma}")
            except ValueError:
                print("Por favor, ingresa solo números enteros.")
        else:
            print("Entrada inválida. Asegúrate de ingresar exactamente dos números enteros.")
            
if __name__ == "__main__":
    main()
